tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -474,77 +474,153 @@ def with_poetry_module(context: PipelineContext, parent_dir: Directory, module_p
+        .with_directory("/airbyte", integration_base.directory("/airbyte"))
         .with_exec(["yum", "install", "-y", "tar", "openssl"])
         .with_exec(["yum", "clean", "all"])
         .with_workdir("/airbyte")
-        .with_file("dd-java-agent.jar", datadog_java_agent)
-        .with_file("javabase.sh", javabase_sh)
+        .with_file("dd-java-agent.jar", context.dagger_client.http("https://dtdg.co/latest-java-tracer"))
+        .with_file("javabase.sh", context.get_repo_dir("airbyte-integrations/bases/base-java", include=["javabase.sh"]).file("javabase.sh"))
         .with_env_variable("AIRBYTE_SPEC_CMD", "/airbyte/javabase.sh --spec")
         .with_env_variable("AIRBYTE_CHECK_CMD", "/airbyte/javabase.sh --check")
         .with_env_variable("AIRBYTE_DISCOVER_CMD", "/airbyte/javabase.sh --discover")
         .with_env_variable("AIRBYTE_READ_CMD", "/airbyte/javabase.sh --read")
         .with_env_variable("AIRBYTE_WRITE_CMD", "/airbyte/javabase.sh --write")
         .with_env_variable("AIRBYTE_ENTRYPOINT", "/airbyte/base.sh")
-        .with_label("io.airbyte.version", java_base_version)
+        .with_label("io.airbyte.version", "0.1.2")
         .with_label("io.airbyte.name", "airbyte/integration-base-java")
     )
 
 
-async def with_airbyte_java_connector(context: ConnectorContext, connector_java_tar_file: File, build_platform: str):
+BASE_DESTINATION_SPECIFIC_NORMALIZATION_DOCKERFILE_MAPPING = {
+    "destination-clickhouse": "clickhouse.Dockerfile",
+    "destination-duckdb": "duckdb.Dockerfile",
+    "destination-mssql": "mssql.Dockerfile",
+    "destination-mysql": "mysql.Dockerfile",
+    "destination-oracle": "oracle.Dockerfile",
+    "destination-tidb": "tidb.Dockerfile",
+    "destination-bigquery": "Dockerfile",
+    "destination-redshift": "redshift.Dockerfile",
+    "destination-snowflake": "snowflake.Dockerfile",
+}
+
+BASE_DESTINATION_SPECIFIC_NORMALIZATION_ADAPTER_MAPPING = {
+    "destination-clickhouse": "dbt-clickhouse>=1.4.0",
+    "destination-duckdb": "duckdb.Dockerfile",
+    "destination-mssql": "dbt-sqlserver==1.0.0",
+    "destination-mysql": "dbt-mysql==1.0.0",
+    "destination-oracle": "dbt-oracle==0.4.3",
+    "destination-tidb": "dbt-tidb==1.0.1",
+    "destination-bigquery": "dbt-bigquery==1.0.0",
+}
+
+DESTINATION_SPECIFIC_NORMALIZATION_DOCKERFILE_MAPPING = {
+    **BASE_DESTINATION_SPECIFIC_NORMALIZATION_DOCKERFILE_MAPPING,
+    **{f"{k}-strict-encrypt": v for k, v in BASE_DESTINATION_SPECIFIC_NORMALIZATION_DOCKERFILE_MAPPING.items()},
+}
+
+DESTINATION_SPECIFIC_NORMALIZATION_ADAPTER_MAPPING = {
+    **BASE_DESTINATION_SPECIFIC_NORMALIZATION_ADAPTER_MAPPING,
+    **{f"{k}-strict-encrypt": v for k, v in BASE_DESTINATION_SPECIFIC_NORMALIZATION_ADAPTER_MAPPING.items()},
+}
+
+
+def with_normalization(context: ConnectorContext) -> Container:
+    normalization_directory = context.get_repo_dir("airbyte-integrations/bases/base-normalization")
+    sshtunneling_file = context.get_repo_dir(
```

Comment:
Nice suggestion, thanks. I think I did not implement it originally because Dagger team told me they would expose such a method in their SDK.
As it will imply unrelated refacto I logged an issue to do it: https://github.com/airbytehq/airbyte/issues/25410